In [1]:
from scipy.optimize import linprog

#Define the objective function
c = [-100,-100,-100,-100,-100,-60,-60,-60,-60,-60,-110,-110,-110,-110,-110] # minimize (-P)

#Define the constraints
cof = ([[1,0,0,0,0,1,0,0,0,0,1,0,0,0,0],[0,1,0,0,0,0,1,0,0,0,0,1,0,0,0],[0,0,1,0,0,0,0,1,0,0,0,0,1,0,0],[0,0,0,1,0,0,0,0,1,0,0,0,0,1,0],[0,0,0,0,1,0,0,0,0,1,0,0,0,0,1],[1.6,0,0,0,0,2.9,0,0,0,0,3.5,0,0,0,0],[0,1.6,0,0,0,0,2.9,0,0,0,0,3.5,0,0,0],[0,0,1.9,0,0,0,0,2.9,0,0,0,0,3.5,0,0],[0,0,0,1.6,0,0,0,0,2.9,0,0,0,0,3.5,0],[0,0,0,0,1.6,0,0,0,0,2.9,0,0,0,0,3.5],[50,50,50,50,50,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,1.5,1.5,1.5,1.5,1.5,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,2.2,2.2,2.2,2.2,2.2],[1.6,1.6,1.6,1.6,1.6,2.9,2.9,2.9,2.9,2.9,3.5,3.5,3.5,3.5,3.5]]) # coefficient matrix 
rhs = ([2000,2300,600,1100,500,3200,3400,800,500,600,110000,1800,2200,7400])

#Define the bounds
b = (0,float('inf')) # x is a positive real number

#Solve
res = linprog(c, A_ub=cof, b_ub=rhs, bounds=(b,b,b,b,b,b,b,b,b,b,b,b,b,b,b))

#Print the results
print("Objective function: ", res.fun)
print("Optimal solution: ", res.x)
res

Objective function:  -337862.0689655172
Optimal solution:  [1603.94736842    0.          221.05263158    0.          375.
    0.            0.          131.03448276    0.            0.
    0.          857.14285714    0.          142.85714286    0.        ]


        message: Optimization terminated successfully. (HiGHS Status 7: Optimal)
        success: True
         status: 0
            fun: -337862.0689655172
              x: [ 1.604e+03  0.000e+00 ...  1.429e+02  0.000e+00]
            nit: 7
          lower:  residual: [ 1.604e+03  0.000e+00 ...  1.429e+02
                              0.000e+00]
                 marginals: [ 0.000e+00  2.842e-14 ...  0.000e+00
                              0.000e+00]
          upper:  residual: [       inf        inf ...        inf
                                    inf]
                 marginals: [ 0.000e+00  0.000e+00 ...  0.000e+00
                              0.000e+00]
          eqlin:  residual: []
                 marginals: []
        ineqlin:  residual: [ 3.961e+02  1.443e+03 ...  0.000e+00
                              0.000e+00]
                 marginals: [-0.000e+00 -0.000e+00 ... -1.708e+01
                             -2.069e+01]
 mip_node_count: 0
 mip_dual_bound: 0.0
        mip_

interpretation of the Model Output

Maximum profit= Rs 337.862 million

Production Plan (in acres)( since the problem has got multiple solutions, there can be few cultivation plans but maximum profit value should be same for every optimum cultivation plan)
||Black gram|Sesame|Big onion|
|-------|-------|-------|-------|
|Thanthirimale|1604|0|0|
|Pulmude|0|0|857|
|Rambewa|221|131|0|
|Tirappane|0|0|143|
|Rajanganaya|375|0|0|


In [2]:
from scipy.optimize import linprog

#Define the objective function
c = [-100,-100,-100,-100,-100,-60,-60,-60,-60,-60,-110,-110,-110,-110,-110] # minimize (-P)

#Define the constraints
cof = ([[1,0,0,0,0,1,0,0,0,0,1,0,0,0,0],[0,1,0,0,0,0,1,0,0,0,0,1,0,0,0],[0,0,1,0,0,0,0,1,0,0,0,0,1,0,0],[0,0,0,1,0,0,0,0,1,0,0,0,0,1,0],[0,0,0,0,1,0,0,0,0,1,0,0,0,0,1],[1.6,0,0,0,0,2.9,0,0,0,0,3.5,0,0,0,0],[0,1.6,0,0,0,0,2.9,0,0,0,0,3.5,0,0,0],[0,0,1.9,0,0,0,0,2.9,0,0,0,0,3.5,0,0],[0,0,0,1.6,0,0,0,0,2.9,0,0,0,0,3.5,0],[0,0,0,0,1.6,0,0,0,0,2.9,0,0,0,0,3.5],[50,50,50,50,50,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,1.5,1.5,1.5,1.5,1.5,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,2.2,2.2,2.2,2.2,2.2],[1.6,1.6,1.6,1.6,1.6,2.9,2.9,2.9,2.9,2.9,3.5,3.5,3.5,3.5,3.5]]) # coefficient matrix 
rhs = ([2000,2300,600,1100,500,3200,3400,800,500,600,110000,1800,2200,8000])

#Define the bounds
b = (0,float('inf')) # x is a positive real number

#Solve
res = linprog(c, A_ub=cof, b_ub=rhs, bounds=(b,b,b,b,b,b,b,b,b,b,b,b,b,b,b))

#Print the results
print("Objective function: ", res.fun)
print("Optimal solution: ", res.x)

Objective function:  -350275.8620689655
Optimal solution:  [1825.            0.            0.            0.          375.
    0.            0.          275.86206897   62.06896552    0.
    0.          908.57142857    0.           91.42857143    0.        ]


interpretation of the Model Output

Maximum profit= Rs (350.276 - 06) millions = 344.276 millions

Production Plan (in acres)( since the problem has got multiple solutions, there can be few cultivation plans but maximum profit value should be same for every optimum cultivation plan)
||Black gram|Sesame|Big onion|
|-------|-------|-------|-------|
|Thanthirimale|1825|0|0|
|Pulmude|0|0|908|
|Rambewa|0|276|0|
|Tirappane|0|62|91|
|Rajanganaya|375|0|0|

compare with the previous solution 

Point to think.......... 
if additional allotment of 600 acre-feet of water for an additional fee of Rs 15,000,000 is obtained, do you recomend 